pkg/propertyprovider/azure/provider.go

```diff
@@ -69,6 +70,9 @@ type PropertyProvider struct {
 	// The controller manager in use by the Azure property provider; this field is mostly reserved for
 	// testing purposes.
 	mgr ctrl.Manager
+
+	// The cloud configuration in use by the Azure property provider.
+	cloudConfig cloudconfig.CloudConfig
```

Comment:
what you need is an azure client. It's better to initialize the client in the main func, so that it can be shared by the whole component.

here we don't need to pass in the whole cloudConfig

Reply:
Would that mean we would initialize all the clients we needed too? I also need armnetwork.NewVirtualNetworksClient